Core/include/Acts/EventData/detail/TestSourceLink.hpp

```diff
@@ -88,12 +88,18 @@ struct TestSourceLink final {
   }
   constexpr std::size_t index() const { return sourceId; }
 
+  template <typename geometry_t>
   struct SurfaceAccessor {
```

Comment:
Would it be possible to alternatively have a `Gen2` `SurfaceAccessor`? I would prefer the code duplication over the templating

Reply:
This was (almost) the initial implementation (see #3300). The implementation seemed kinda clunky, though. Also, removing templates in the perspective when the geometry is persistified sounds easier than looking for the duplicates. @asalzburger, what do you think?